x/gamm/simulation/sim_msgs.go

```diff
@@ -472,3 +412,24 @@ func createPoolRestriction(k keeper.Keeper, sim *simtypes.SimCtx, ctx sdk.Contex
 		return hasTwoCoins && hasPoolCreationFee
 	}
 }
+
+func getRandPool(k keeper.Keeper, sim *simtypes.SimCtx, ctx sdk.Context) (uint64, types.PoolI, sdk.Coin, sdk.Coin, []string, string, error) {
+	// select a pseudo-random pool ID, max bound by the upcoming pool ID
+	pool_id := simtypes.RandLTBound(sim, k.GetNextPoolNumber(ctx))
+	pool, err := k.GetPoolAndPoke(ctx, pool_id)
+	if err != nil {
+		return 0, nil, sdk.NewCoin("denom", sdk.ZeroInt()), sdk.NewCoin("denom", sdk.ZeroInt()), []string{}, "", err
+	}
+	poolCoins := pool.GetTotalPoolLiquidity(ctx)
+
+	// TODO: Improve this, don't just assume two asset pools
+	// randomly select one of the pool denoms to be the coinIn and one to be the coinOut
+	r := sim.GetSeededRand("select random seed")
+	index := r.Intn(len(poolCoins) - 1)
+	coinIn := poolCoins[index]
+	poolCoins = simtypes.RemoveIndex(poolCoins, index)
+	coinOut := poolCoins[0]
```

Comment:
Why not use route directly here?

Reply:
The route various based on the type right? SwapAmountInRoute vs SwapAmountOutRoute, they are different types so couldn't use the same output AFAIU

Reply:
ah right :/

Reply:
And because there is only two occurrences of this, creating two separate functions for each type wouldn't make sense right?

Reply:
I'm just making sure I am not missing something! 😄